Implement Map View

johnskimca opened the pull request:
#### [Title](#)
Implemented Map view for volunteers to identify location of stores that need help

**What and how**

1. Utilized 'react-google-maps' library to render google maps view centered around Vancouver - _Maps.js in components_
2. Implemented markers (using data from constant.js) to show businesses who need help - _Maps.js in components_
3. Implemented an info view which opens with the name of store, address, and avatar (picture) of the store - _Maps.js in components_
4. Created styles for the Maps component (for both full screen and small screen)- _maps.module.css in modules_
5. Installed switch button component on volunteers navbar (in both small screen and full screen mode) - _Switch.js in NavBar Component and NavBar in components_
5. Added action (switchView) and reducer (switchView) to track the state of the switch (either maps view or list view) - _index.js in action and reducers_ 

**Screenshots**

![image](https://user-images.githubusercontent.com/59780084/85046637-8aeeff80-b145-11ea-9c8e-4c8ee89272e0.png)

![image](https://user-images.githubusercontent.com/59780084/85046688-9c380c00-b145-11ea-9284-3bd20e85f73b.png)

![image](https://user-images.githubusercontent.com/59780084/85046728-a8bc6480-b145-11ea-8827-e4da543200d7.png)

![image](https://user-images.githubusercontent.com/59780084/85046799-c5f13300-b145-11ea-8bc4-f10d77e523ae.png)

![image](https://user-images.githubusercontent.com/59780084/85046833-d1445e80-b145-11ea-9421-331f92899fb2.png)

![image](https://user-images.githubusercontent.com/59780084/85046881-e3260180-b145-11ea-9afa-de56c5a84f9c.png)

johnskimca committed:
Completed Navbar:

buttonNavBar.js - Created seperate button component (conditional rendering based on props)

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into navBar

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into navBar

johnskimca committed:
Completed Navbar:

before pull

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into navBar

 Conflicts:
	client/src/App.js
	client/src/components/NavBar.js

johnskimca committed:
Fixed issue with API Key -> still having trouble with .env.local

johnskimca committed:
.env.local finished and switch added but cut off

johnskimca committed:
Switch installed, redux set up, and code in map cleaned up

johnskimca committed:
Dispatch is working friend!!

johnskimca committed:
Map vs List View implemented

johnskimca committed:
Pictures added to the infoWindow and resizes based on size of screen.

However, I need help on the following issue: when the screen resizes (goes to small hamburger view or moves out from small screen hamburger view to full screen) the switch resets to list position. It does not remember what switch state it was in before. This means switching screen size can fool our switch to state we are in listview when we are actually in map view.

I'm not sure how I can fix this, unless resizing dispatches an action that is always false (list)

yehee commented:
Love the toggle on the nav bar! Can you run the linter? Thanks!

johnskimca committed:
ran:

npm run lint: which caused 5 warnings to appear.
npm run lint:fix (warnings don't appear to be resolved, just errors)

Manually resolved the warning that appears in Switch.js

johnskimca commented:
> Love the toggle on the nav bar! Can you run the linter? Thanks!

Thank you, I ran npm run lint which produced 5 warnings, 1 of which pertained to switch. 

I resolved that issue manually. There are 4 remaining warnings that do not pertain to the components that I worked on. Just pushed on my latest commit.

![image](https://user-images.githubusercontent.com/59780084/85072362-dc12e980-b16d-11ea-85ce-a2b917fe649f.png)

yehee commented on client/src/components/Maps.js:
```diff
@@ -0,0 +1,82 @@
+    const [selectedBusiness, setSelectedBusiness] = useState(null);
+    const Markers = () => (
+        <div>
+            {businesses.map(business => (business.needsHelp
```
Could use [this](http://es6-features.org/#ObjectMatchingShorthandNotation)
i.e.
```javascript
{businesses.map(({ id, lat, lng, needsHelp }) => (needsHelp
```

johnskimca replied:
I am having some issues with changing to object destructuring for 2 reasons.

**Reason 1**:
business to  {id, lat, lng, needsHelp } => only gives me access to those 4 fields. However, down in InfoWindows, I also require access to storeName, street address and avatar. This would mean that object destructuring would be 7 fields. 

**Reason 2**
Lint is asking that I reorganize the new object destructured function into something that is more difficult to read. 

If okay, I would like to keep this the same. However, if you feel strongly about this, please let me know and I will try to change.

yehee replied:
As I said earlier in the Slack, let's not worry about it 👍 Thanks for the heads up!

yehee commented on client/src/components/NavBar/components/Switch.js:
```diff
@@ -0,0 +1,78 @@
+
+
+    const handleChange = event => {
+        // setState({ ...state, checkedC: switchState });
```
Could be cleaned up

johnskimca replied:
Completed! All comments cleaned up.

yehee commented on client/src/reducers/index.js:
```diff
@@ -124,4 +126,13 @@ export default combineReducers({
                 return conversation;
         }
     },
+    switchView: (initialSwitchState = switchState, action) => {
```
**NITPICKING**, but could call `initialSwitchState` -> `switchState`, and `switchState` -> `defaultSwitchState` to abide with the general convention.

johnskimca replied:
Done!

yehee commented on client/src/reducers/index.js:
```diff
@@ -124,4 +126,13 @@ export default combineReducers({
     },
+    switchView: (initialSwitchState = switchState, action) => {
+        switch (action.type) {
+            case SWITCH_VIEW:
+                const updatedSwitchState = !initialSwitchState;
+                return updatedSwitchState;
```
```suggestion
            case SWITCH_VIEW:
                return  !initialSwitchState;
```

yehee commented on package.json:
```diff
@@ -9,5 +9,8 @@
     "lint:fix": "cd client && yarn lint:fix"
+  },
+  "dependencies": {
+    "no-package-lock": "0.0.1"
```
Just what we needed 😭

johnskimca replied:
YES!!

johnskimca committed:
Update client/src/reducers/index.js

Co-authored-by: Alice <aliceykim0828@gmail.com>

johnskimca committed:
Addressed review notes.

Ran lint and fixed error once more.

Only review note not resolved is in regards to object destructuring.

Please read comment and let me know if this ok

johnskimca committed:
Merge remote-tracking branch 'origin/mapsView' into mapsView

# Conflicts:
#	client/src/reducers/index.js

yehee reviewed (approved):
LGTM :shipit:
